packages/preview-middleware-client/src/cpe/changes/service.ts

```diff
@@ -307,4 +322,44 @@ export class ChangeService {
 
         return result;
     }
+
+    private getCommandChangeType(command: FlexCommand): string | undefined {
```

Comment:
Here we have a duplication with the try-catch, I am suggesting making those operations more generic:

```typescript
private attemptOperations<T>(operations: Array<() => T>): T | undefined {
    for (const operation of operations) {
        try {
            return operation();
        } catch (error) {
            // Log error
            // Continue to the next operation
        }
    }
    return undefined;
}

private getCommandChangeType(command: FlexCommand): string | undefined {
    return this.attemptOperations([
        () => command.getChangeType(),
        () => command.getPreparedChange().getDefinition().changeType
    ]);
}

private getCommandSelectorId(command: FlexCommand): string | undefined {
    return this.attemptOperations([
        () => command.getSelector().id,
        () => command.getElement().getProperty('persistencyKey')
    ]);
}
```

This way we have less code and consistent return in case of an error.